spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/BasicRelationalPersistentProperty.java

```diff
@@ -88,14 +91,26 @@ public BasicRelationalPersistentProperty(Property property, PersistentEntity<?,
 		this.columnName = Lazy.of(() -> Optional.ofNullable( //
 				findAnnotation(Column.class)) //
 				.map(Column::value) //
-				.filter(StringUtils::hasText) //
+				.filter(StringUtils::hasText)//
 		);
 
 		this.keyColumnName = Lazy.of(() -> Optional.ofNullable( //
```

Comment:
This code should be inlined into `collectionKeyColumnName` resolution and the field should be removed.